src/wallet/wallet.ts

```diff
@@ -65,35 +92,47 @@ class HathorWalletServiceWallet extends EventEmitter {
     this.walletId = null;
     this.isSendingTx = false;
     this.txProposalId = null;
+    this.xpub = null;
 
     this.network = network;
+
+    this.authToken = null;
+
+    // TODO When we integrate the real time events from the wallet service
+    // we will need to have a trigger to update this array every new transaction
+    // because the new tx might have used one of those addresses
+    // so we just need to call await this.getNewAddresses(); on the new tx event
+    this.newAddresses = [];
+    this.indexToUse = -1;
     // TODO should we have a debug mode?
   }
 
   /**
    * Start wallet: load the wallet data, update state and start polling wallet status until it's ready
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
   async start() {
     this.setState(walletState.LOADING);
-    const xpub = wallet.getXPubKeyFromSeed(this.seed, {passphrase: this.passphrase});
-    const handleCreate = async (data) => {
+    const xpub = wallet.getXPubKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
+    this.xpub = xpub;
+    const handleCreate = async (data: WalletStatus) => {
       this.walletId = data.walletId;
       if (data.status === 'creating') {
         await this.startPollingStatus();
       } else {
-        this.setState(walletState.READY);
+        await this.onWalletReady();
       }
     }
+
     try {
-      const res = await walletApi.createWallet(xpub);
+      const res = await walletApi.createWallet(this.xpub);
       const data = res.data;
```

Comment:
Shouldn't we move this logic to the walletApi? I think this code must be way simpler to read because it would have to deal with a promise that might succeed or fail.

Reply:
Makes sense, done.

Reply:
This refactor is great. It is way easier to read the code. Thanks!